cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/Executor.java

```diff
@@ -607,30 +605,27 @@ private void moveReplicas() {
         inExecutionTasks = _executionTaskManager.inExecutionTasks();
       }
-      if (_executionTaskManager.inProgressTasks().isEmpty()) {
-        LOG.info("Partition movements finished.");
-      } else if (_stopRequested.get()) {
-        ExecutionTaskManager.ExecutionTasksSummary executionTasksSummary = _executionTaskManager.getExecutionTasksSummary();
-        LOG.info("Partition movements stopped. {} in-progress, {} pending, {} aborting, {} aborted, {} dead, "
-                 + "{} remaining data to move.",
-                 executionTasksSummary.inProgressTasks().size(),
-                 executionTasksSummary.remainingPartitionMovements().size(),
-                 executionTasksSummary.abortingTasks(),
-                 executionTasksSummary.abortedTasks().size(),
-                 executionTasksSummary.deadTasks().size(),
-                 executionTasksSummary.remainingDataToMoveInMB());
-      }
+      ExecutionTasksSummary executionTasksSummary = _executionTaskManager.getExecutionTasksSummary(Collections.emptyList(), false);
+      LOG.info("Partition movements {} with {} tasks {}, {} tasks in-progress, {} tasks aborting, {} tasks aborted, {} tasks dead, {} tasks completed.",
+               _stopRequested.get() ? STOPPED : FINISHED,
+               executionTasksSummary.taskStat().get(REPLICA_ACTION).get(ExecutionTask.State.PENDING),
+               _stopRequested.get() ? CANCELLED : PENDING,
+               executionTasksSummary.taskStat().get(REPLICA_ACTION).get(ExecutionTask.State.IN_PROGRESS),
+               executionTasksSummary.taskStat().get(REPLICA_ACTION).get(ExecutionTask.State.ABORTING),
+               executionTasksSummary.taskStat().get(REPLICA_ACTION).get(ExecutionTask.State.ABORTED),
+               executionTasksSummary.taskStat().get(REPLICA_ACTION).get(ExecutionTask.State.DEAD),
+               executionTasksSummary.taskStat().get(REPLICA_ACTION).get(ExecutionTask.State.COMPLETED));
     }
 
     private void moveLeaderships() {
-      int numTotalLeadershipMovements = _executionTaskManager.remainingLeadershipMovements().size();
+      int numTotalLeadershipMovements = _executionTaskManager.numRemainingLeadershipMovements();
       LOG.info("Starting {} leadership movements.", numTotalLeadershipMovements);
-      _numFinishedLeadershipMovements = 0;
-      while (!_executionTaskManager.remainingLeadershipMovements().isEmpty() && !_stopRequested.get()) {
+      while (_executionTaskManager.numRemainingLeadershipMovements() != 0 && !_stopRequested.get()) {
         updateOngoingExecutionState();
-        _numFinishedLeadershipMovements += moveLeadershipInBatch();
-        LOG.info("{}/{} ({}%) leadership movements completed.", _numFinishedLeadershipMovements,
-                 numTotalLeadershipMovements, _numFinishedLeadershipMovements * 100 / numTotalLeadershipMovements);
+        moveLeadershipInBatch();
+        int numFinishedLeadershipMovements = _executionTaskManager.numFinishedLeadershipMovements();
```

Comment:
It's not clear why we would make this change -- i.e. wouldn't this change receive the same information in a more expensive manner?

Previously we were just using the return value of `moveLeadershipInBatch()`, now we are making an expensive computation to get the same value.